File: Proxy/src/main/kotlin/DynamicProxy/Jdk/JdkEx/service/impl/SomeServiceImpl.java
package DynamicProxy.Jdk.JdkEx.service.impl;


import DynamicProxy.Jdk.JdkEx.service.SomeService;

// service类的代码不修改，也能够增加 输出时间， 事务。
public class SomeServiceImpl implements SomeService {
    @Override
    public void doSome() {
        System.out.println("执行业务方法doSome");
        int res  = 10 + 20;
        //更新购买商品的库存， 生成订单， 结算整个购买商品的总价。
    }

    @Override
    public void doOther() {
        System.out.println("执行业务方法doOther");
    }
}

File: Proxy/src/main/kotlin/DynamicProxy/Aspect/Service/SomeServiceImpl.kt
package DynamicProxy.Aspect.Service

import DynamicProxy.Aspect.model.User
import org.springframework.stereotype.Component

//通过spring创建对象
@Component("someService")
class SomeServiceImpl : SomeService {
    override fun doSome(name: String, age: Int) {
        //之前输出时间
        println(" -----目标doSome---- ")
        println("$name : $age")

    }

    override fun doOther(name: String, age: Int): String {
        println(" -----目标doOther---- ")
        return "$name : $age"
    }

    override fun doUser(name: String, age: Int): User {
        println(" -----目标doUser---- ")
        return User(name, age)
    }

    override fun doAround(name: String, age: Int): User {
        println(" -----目标doUser---- ")
        return User(name, age)

    }

    override fun doAfterThrowing(x: Int, y: Int) {
        println(x / y)
    }
}